test/core/runtime/binaryen/wasm_memory_test.cpp

```diff
@@ -34,8 +37,12 @@ class BinaryenMemoryHeapTest : public ::testing::Test {
     rei_ =
         std::make_unique<runtime::binaryen::RuntimeExternalInterface>(host_api);
 
-    memory_ = std::make_unique<MemoryImpl>(
-        rei_->getMemory(), runtime::MemoryConfig{kDefaultHeapBase, {}});
+    runtime::MemoryConfig config{kDefaultHeapBase, {}};
+    auto handle = std::make_shared<MemoryImpl>(rei_->getMemory(), config);
+    auto allocator =
+        std::make_unique<runtime::MemoryAllocatorImpl>(handle, config);
+    allocator_ = allocator.get();
+    memory_ = std::make_unique<runtime::Memory>(handle, std::move(allocator));
```

Comment:
How about combine calls for Memory assemble into single free-function?

Reply:
I thought about it, but in the end it didn't seem to be very useful, as it would eliminate just a couple of simple lines of code that create allocator across the entire code base.